source/Calamari/Commands/KubernetesObjectStatusReporterCommand.cs

```diff
@@ -0,0 +1,95 @@
+#if !NET40
+using System;
+using System.Collections.Generic;
+using Calamari.Common.Commands;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Proxies;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Kubernetes;
+using Calamari.Kubernetes.Integration;
+using Calamari.Kubernetes.ResourceStatus;
+
+namespace Calamari.Commands
+{
+    [Command("kubernetes-object-status")]
+    public class KubernetesObjectStatusReporterCommand: Command<KubernetesObjectStatusReporterCommandInput>
+    {
+        private readonly ILog log;
+        private readonly IVariables variables;
+        private readonly IResourceStatusReportExecutor statusReportExecutor;
+        private readonly Kubectl kubectl;
+
+        public KubernetesObjectStatusReporterCommand(
+            ILog log,
+            IVariables variables,
+            IResourceStatusReportExecutor statusReportExecutor,
+            Kubectl kubectl)
+        {
+            this.log = log;
+            this.variables = variables;
+            this.statusReportExecutor = statusReportExecutor;
+            this.kubectl = kubectl;
+        }
+
+        private bool IsEnabled()
+        {
+            var resourceStatusEnabled = variables.GetFlag(SpecialVariables.ResourceStatusCheck);
```

Comment:
@YihaoOct wondering if some of these props can be typed inputs?
Example:
-  "Octopus.Action.Kubernetes.ResourceStatusCheck"
-  "Octopus.Action.Kubernetes.DeploymentTimeout"
-   "Octopus.Action.KubernetesContainers.DeploymentStyle" <- Maybe this one does not make sense?
-  "Octopus.Action.KubernetesContainers.DeploymentWait" <- This one is for backward compatibility right, so not needed?
-  "Octopus.Action.Kubernetes.WaitForJobs"

So the idea is to have:
```c#
public class KubernetesObjectStatusReporterCommandInput
{
    public bool WaitForJobs { get; set; }
    public int DeploymentTimeout { get; set; }
    public bool Enabled { get; set; }
}
```

What do you think?

Reply:
I ended up going down this path anyway